src/DnsTools.Worker/Tools/BaseCliTool.cs

```diff
@@ -32,90 +29,61 @@ public abstract class BaseCliTool<TRequest, TResponse> : ITool<TRequest, TRespon
 			CancellationToken cancellationToken
 		)
 		{
+			var args = await GetArguments(request, responseStream);
 			var wrap = Cli.Wrap(GetCommand(request))
-				.SetArguments(await GetArguments(request, responseStream))
-				.EnableExitCodeValidation(false)
-				.EnableStandardErrorValidation(false);
+				.WithArguments(x => x.Add(args))
```

Comment:
Note, if your `args` is a fully-formed arguments string, you will want to use `.WithArguments(args)`. Using `.Add()` will treat each item as a separate argument and will escape spaces, quotes, etc in them accordingly.

Reply:
@Tyrrrz `args` is a list of arguments, for example I have this code elsewhere:

```csharp
		protected override async Task<IReadOnlyList<string>> GetArguments(
			PingRequest request,
			IServerStreamWriter<PingResponse> writer
		)
		{
			var ip = await Hostname.GetIp(request.Host, request.Protocol);
			var ipString = ip.ToString();
			await writer.WriteAsync(new PingResponse
			{
				Lookup = new HostLookupResult
				{
					Ip = ipString
				}
			});

			return new List<string> { "-i 0.5", "-c 5", "-O", ipString };
		}
```

What do you think about adding a `WithArguments(IEnumerable<string>)` overload that just calls `WithArguments(x => x.Add(args))`?

Reply:
Ah, sorry, I didn't realize it was a list! Then it's going to work as expected.

> What do you think about adding a WithArguments(IEnumerable<string>) overload that just calls WithArguments(x => x.Add(args))?

That sounds like a good idea.

Reply:
Actually, not exactly as expected. I can see you're returning

```csharp
return new List<string> { "-i 0.5", "-c 5", "-O", ipString };
```

In this case, `-i 0.5` and `-c 5` are treated as two arguments instead of four. The resulting string will be something like:

```
"-i 0.5" "-c 5" -O
```

With the quotes escaping spaces within arguments. Depending on the implementation of the CLI, it might treat these incorrectly.

You'll want something like this instead:

```csharp
return new List<string> { "-i", "0.5", "-c", "5", "-O", ipString };
```

Is this behavior confusing?

Reply:
@Tyrrrz Hmm, good point. It's working fine though, at least on Linux for the commands I'm using. 

I think semantically, I'd say that `-i 0.5` is a single parameter, as `0.5` is a value being passed to the `-i` option. Different parsers parse it differently though. Command line parsing is definitely a bit of a mess.

Reply:
Yeah there are different semantics surrounding arguments, parameters, options. In general, arguments are the individual space-separated items, like the ones you receive in `Main(string[] args)`.